receiver/saphanareceiver/client.go

```diff
@@ -123,9 +122,13 @@ func (c *sapHanaClient) Connect(ctx context.Context) error {
 	if err != nil {
 		return fmt.Errorf("error generating TLS config for SAP HANA connection: %w", err)
 	}
-	connector.SetTLSConfig(tls)
+	if err = connector.SetTLSConfig(tls); err != nil {
+		return fmt.Errorf("error setting TLS config for SAP HANA connection: %w", err)
+	}
 
-	connector.SetApplicationName("OpenTelemetry Collector")
+	if err = connector.SetApplicationName("OpenTelemetry Collector"); err != nil {
+		return fmt.Errorf("error setting application name for SAP HANA connection: %w", err)
```

Comment:
@dehaansa, do you think it is correct to return an error in these cases?

Reply:
I think it's reasonable. If you're unable to do something simple like set the application name most likely something is quite wrong with the connector.